feat: Weapons Station

joachimboyden opened the pull request:
Created a weapons station prefab (by default sells a shotgun) and a shotgun variant. 
Since there is no point system yet, I've made both the weapon and the ammo free for now. I've also made the ammo refill all your weapons ammo, but not the clip its self.

There are no Game Scene changes, so to test, just drag the prefab in. 
Also created a Interactable class, it currently only takes a radius of interaction and it's materials.

JDBryan reviewed (changes requested):
If I buy a shotgun from the weapon station and then lose the shotgun by picking up another weapon, the weapon station will still only allow me to buy ammo rather than the expected behaviour of purchasing a weapon

joachimboyden commented:
Fixed your problem with dropping weapons and some other bugs I noticed. I also dragged the shotgun station into the game scene and deleted the shotgun that was on the ground.

JDBryan commented on Assets/Scripts/Interactable.cs:
```diff
@@ -0,0 +1,36 @@
+        BaseStart();
+    }
+
+    public void BaseStart(){
```
Rename function to Init

JDBryan commented on Assets/Scripts/WeaponStation.cs:
```diff
@@ -0,0 +1,64 @@
+            this.restockInProgress = true;
+            animator.SetBool("Reload", true);
+        }
+        if (this.weaponOnSale == false & this.restockInProgress == false){
```
Feels like you should maybe be using an else if here?

JDBryan reviewed (approved):
Approved with a couple of small comments
